daffodil-runtime1/src/main/scala/org/apache/daffodil/api/DFDLParserUnparser.scala

```diff
@@ -226,84 +221,22 @@ object DFDL {
   }
 
   trait DaffodilUnparseContentHandler
-    extends  ProducerCoroutine
-    with org.xml.sax.ContentHandler {
+    extends org.xml.sax.ContentHandler {
     def getUnparseResult: UnparseResult
-    def enableInputterResolutionOfRelativeInfosetBlobURIs(): Unit
+    def enableResolutionOfRelativeInfosetBlobURIs(): Unit
   }
```

Comment:
The trait acts as an interface so that if this .api ever does become public (which I hope happens some day, it's such a pain having to maintain daffodil-japi and daffodil-sapi) then this will be the public interface that users can interact with. But I can definitely see how that could be confusing. Maybe we rename the implementation to `DaffodilUnparseContentHandlerImpl` to make it clear it's an implementation of the DaffodilUnparseContentHandler interface/API.

I'm not sure if we have a standard on fully qualified vs not, but in general I prefer to not use the fully qualified name, I guess I missed a couple places.

Reply:
"Impl" suffix helps here. 
I like to use "Mixin" suffix also when a trait provides implementation methods. 
In fact both makes sense sometimes XYZImplMixin is useful when there is an interface XYZ. 

As for qualified names,... IDEs make it hard to maintain a discipline of say, not importing a particular package because you *want* to use qualified names.

The IDEs I've used all prompt you for possible matching names, showing you the choices for different qualifiers, and then insert the name in the code and the import statement at the top for you. 

This is a big help when writing code and I would *not* want to turn it off for the sake of some discretionary code style issue like "I think I'd prefer to use a qualified name in this spot."  If you really insist on this you can put in the qualified name and delete the import and the IDE won't undo your change, but you have to really want it that way to go through those steps.